docs/tutorials/embedding/MASE.ipynb

```diff
@@ -0,0 +1,374 @@
+   "source": [
+    "## Visualize Common Latent Subspace"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since the two graphs have clear block structures, with higher within-block probabilities than between-block, we see two distinct \"clusters\" when we visualize the latent positions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig, ax = plt.subplots(figsize=(10,10))\n",
+    "ax.scatter(V_hat[:25, 0], V_hat[:25, 1], c=\"blue\", label=\"Block 1\")\n",
+    "ax.scatter(V_hat[25:, 0], V_hat[25:, 1], c=\"red\", label=\"Block 2\")\n",
+    "ax.legend()\n",
+    "ax.set_title(\"Common Latent Subspace from MASE Embedding\", fontsize=20)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Recover Probability Matrices"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since ``MultipleASE()`` stores estimates of the individual score matrices $\\hat{R}^{(i)}$ as well as the common invariant subspace $\\hat{V}$, we can recover an estimate of the probability matrix for each graph \n",
+    "$$\\hat{P}^{(i)} = V R^{(i)} V^T$$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "R = embedder.scores_\n",
+    "R_1 = R[0]\n",
+    "R_2 = R[1]\n",
+    "\n",
+    "V_hat_T = np.transpose(V_hat)\n",
+    "P_1_hat = np.matmul(np.matmul(V_hat, R_1), V_hat_T)\n",
+    "P_2_hat = np.matmul(np.matmul(V_hat, R_2), V_hat_T)\n",
+    "\n",
+    "heatmap(P_1_hat, figsize=(7,7), title=\"Visualization of Probability Matrix 1\")\n",
+    "heatmap(P_2_hat, figsize=(7,7), title=\"Visualization of Probability Matrix 2\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Recover Original Block Probabilities"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we can see the clear block structure once again.  In order to obtain an estmate of the original block probabilities, we can average across the blocks."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "P1_hat_block = np.array([[np.mean(P_1_hat[:25, :25]), np.mean(P_1_hat[:25, 25:])],\n",
+    "               [np.mean(P_1_hat[25:, :25]), np.mean(P_1_hat[25:, 25:])]])\n",
+    "P2_hat_block = np.array([[np.mean(P_2_hat[:25, :25]), np.mean(P_2_hat[:25, 25:])],\n",
+    "               [np.mean(P_2_hat[25:, :25]), np.mean(P_2_hat[25:, 25:])]])\n",
+    "print(\"P1_hat_block =\\n\", P1_hat_block)\n",
+    "print(\"P2_hat_block =\\n\", P2_hat_block)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    " Although we don't get exactly the same block probabilities as we input into the SBM, it's fairly close with only 1 graph for each!"
```

Comment:
"only 1 graph" doesn't mean much, it' the number of vertices that matters much more, because of the clt. if you had 2 graphs with 10000 vertices graphs, your estimates would be much more accurate than if you had 4 graphs with 100 vertices each.

Reply:
good point, I was thinking about the error in estimating the latent subspace, which would only depend on the number of graphs.  But the block probabilities depend on the number of vertices... I will mention it.

Reply:
could you quantify 

> error in estimating the latent subspace 

?

Reply:
bump